File: frontend/src/types/courseContent.ts
/*
 * Copyright (c) 2026 Anshul Negi
 * GitHub: https://github.com/NegiCoder
 * Unauthorized copying, modification, or distribution of this file
 * without explicit permission is prohibited.
 */

import type { ContentType } from '@/types/enums'

export type CourseContentResponse = {
  courseContentId: number
  courseId: number
  courseTitle: string
  moduleName: string
  contentType: ContentType
  urlOrPath: string
  sortOrder: number
  isActive: boolean
  createdAt: string
  updatedAt?: string | null
}

export type CreateCourseContentRequest = {
  courseId: number
  moduleName: string
  contentType: ContentType
  urlOrPath: string
  sortOrder: number
  isActive: boolean
}

export type UpdateCourseContentRequest = {
  moduleName: string
  contentType: ContentType
  urlOrPath: string
  sortOrder: number
  isActive: boolean
}

File: backend/TrainingInstitute.Api/Models/CourseContent.cs
/*
 * Copyright (c) 2026 Anshul Negi
 * GitHub: https://github.com/NegiCoder
 * Unauthorized copying, modification, or distribution of this file
 * without explicit permission is prohibited.
 */

using TrainingInstitute.Api.Models.Enums;

namespace TrainingInstitute.Api.Models;

// course ka ek module / lesson - video, pdf ya link ho sakta hai
public class CourseContent
{
    public int CourseContentId { get; set; }

    public int CourseId { get; set; }

    public Course? Course { get; set; }

    public string ModuleName { get; set; } = string.Empty;

    public ContentType ContentType { get; set; }

    // file ka path ya URL - yahi se khulega
    public string UrlOrPath { get; set; } = string.Empty;

    // UI me kis order me dikhana hai
    public int SortOrder { get; set; }

    public bool IsActive { get; set; } = true;

    public DateTime CreatedAt { get; set; }

    public int? CreatedBy { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public int? UpdatedBy { get; set; }

    // ye module kis kis student ne complete kiya
    public ICollection<StudentModuleProgress> ModuleProgressRows { get; set; } = new List<StudentModuleProgress>();
}
